pkg/proxy/conntrack/conntrack.go

```diff
@@ -30,10 +30,36 @@ import (
 	utilnet "k8s.io/utils/net"
 )
 
-// Utilities for dealing with conntrack
+// Interface for dealing with conntrack
+type Interface interface {
+	// ClearEntriesForIP deletes conntrack entries for connections of the given
+	// protocol, to the given IP.
+	ClearEntriesForIP(ip string, protocol v1.Protocol) error
+
+	// ClearEntriesForPort deletes conntrack entries for connections of the given
+	// protocol and IP family, to the given port.
+	ClearEntriesForPort(port int, isIPv6 bool, protocol v1.Protocol) error
+
+	// ClearEntriesForNAT deletes conntrack entries for connections of the given
+	// protocol, which had been DNATted from origin to dest.
+	ClearEntriesForNAT(origin, dest string, protocol v1.Protocol) error
+
+	// ClearEntriesForPortNAT deletes conntrack entries for connections of the given
+	// protocol, which had been DNATted from the given port (on any IP) to dest.
+	ClearEntriesForPortNAT(dest string, port int, protocol v1.Protocol) error
+}
+
+// execCT implements Interface by execing the conntrack tool
+type execCT struct {
+	execer exec.Interface
+}
 
-// NoConnectionToDelete is the error string returned by conntrack when no matching connections are found
-const NoConnectionToDelete = "0 flow entries have been deleted"
+func NewExec(execer exec.Interface) Interface {
```

Comment:
this should be just `New()` , no? to use it as `conntrack.New(execer)`

Reply:
I was preparing for `NewNetlink()` :slightly_smiling_face: ...

Reply:
ie, this isn't "the standard `conntrack.Interface`", it's "the `conntrack.Interfce` based on exec". Which happens to be the only non-fake interface currently, but that doesn't mean it's necessarily what you should consider "the standard"

Reply:
oh, that sounds great